lib/iris/common/metadata.py

```diff
@@ -1338,39 +1338,14 @@ def equal(self, other, lenient=None):
         return super().equal(other, lenient=lenient)
 
 
-def metadata_manager_factory(cls, **kwargs):
-    """
-    A class instance factory function responsible for manufacturing
-    metadata instances dynamically at runtime.
-
-    The factory instances returned by the factory are capable of managing
-    their metadata state, which can be proxied by the owning container.
-
-    Args:
-
-    * cls:
-        A subclass of :class:`~iris.common.metadata.BaseMetadata`, defining
-        the metadata to be managed.
-
-    Kwargs:
-
-    * kwargs:
-        Initial values for the manufactured metadata instance. Unspecified
-        fields will default to a value of 'None'.
-
-    """
-
+@lru_cache(maxsize=None)
```

Comment:
Explicitly set `maxsize=None` to allow `functools.lru_cache` to work for both [py37](https://docs.python.org/3.7/library/functools.html#functools.lru_cache) and [py38](https://docs.python.org/3.8/library/functools.html#functools.lru_cache)